Add test coverage for: Add certificate info to WKFrameInfo accessor of _WKFocusedElementInfo.

arenevier opened the pull request:
#### 98ce31a30a82ddaa0dbf56e1a9d0e0ea9f4b4397
<pre>
Add test coverage for: Add certificate info to WKFrameInfo accessor of _WKFocusedElementInfo.
<a href="https://bugs.webkit.org/show_bug.cgi?id=297061">https://bugs.webkit.org/show_bug.cgi?id=297061</a>
<a href="https://rdar.apple.com/157669975">rdar://157669975</a>

Reviewed by Abrar Rahman Protyasha.

Api test checking that in
setFocusRequiresStrongPasswordAssistanceHandler callback, _serverTrust
is set correctly in _WKFocusedElementInfo.frame.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::focusedElementInformation):
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
(TestWebKitAPI::TEST(KeyboardInputTests, TestFrameInfoServerTrustDuringStrongPasswordAssistance)):

Canonical link: <a href="https://commits.webkit.org/298455@main">https://commits.webkit.org/298455@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0aa75dd18c42b439dbf2b6e895eb7678f0a2a9e1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115532 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35201 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25718 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121594 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/66077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bae5558e-e0fe-4675-83de-fe28464b714b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117421 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35878 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43780 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87767 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/66077 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/855e10e1-5fd6-46c9-987d-bf91e0d1aad3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118480 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28615 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103687 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68160 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27764 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21804 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65266 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97998 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21925 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124754 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42443 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31805 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/96537 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42810 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99878 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96323 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24512 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41561 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19420 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42328 "Built successfully") | | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41821 "Built successfully") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45150 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43537 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+
+    auto inputDelegate = adoptNS([[TestInputDelegate alloc] init]);
+
+    [inputDelegate setFocusStartsInputSessionPolicyHandler:^_WKFocusStartsInputSessionPolicy(WKWebView *, id <_WKFocusedElementInfo>) {
```
I think this is the style guide issue?
```suggestion
    [inputDelegate setFocusStartsInputSessionPolicyHandler:^_WKFocusStartsInputSessionPolicy(WKWebView *, id<_WKFocusedElementInfo>) {
```

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+        { "/"_s, { "<script>alert('done')</script><input type='password' id='input'>"_s } },
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    auto navigationDelegate = adoptNS([TestNavigationDelegate new]);
```
To adhere to our safer CPP guidelines
```suggestion
    RetainPtr navigationDelegate = adoptNS([TestNavigationDelegate new]);
```

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+    [navigationDelegate allowAnyTLSCertificate];
+
+    RetainPtr configuration = server.httpsProxyConfiguration();
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
```
Ditto
```suggestion
    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
```

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
+    webView.get().navigationDelegate = navigationDelegate.get();
+
+    auto inputDelegate = adoptNS([[TestInputDelegate alloc] init]);
```
Ditto
```suggestion
    RetainPtr inputDelegate = adoptNS([[TestInputDelegate alloc] init]);
```

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+        return _WKFocusStartsInputSessionPolicyAllow;
+    }];
+
+    [inputDelegate setFocusRequiresStrongPasswordAssistanceHandler:[&] (WKWebView *, id<_WKFocusedElementInfo> info, void(^completionHandler)(BOOL)) {
```
style guide, no space between capture and arg list
```suggestion
    [inputDelegate setFocusRequiresStrongPasswordAssistanceHandler:[&](WKWebView *, id<_WKFocusedElementInfo> info, void(^completionHandler)(BOOL)) {
```

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+
+    RetainPtr configuration = server.httpsProxyConfiguration();
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
+    webView.get().navigationDelegate = navigationDelegate.get();
```
Optional nit: I think we prefer
```suggestion
    [webView setNavigationDelegate:navigationDelegate.get()];
```

aprotyas reviewed (approved):
LGTM, just style comments

aprotyas commented on Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
```diff
@@ -1524,6 +1526,39 @@ static BOOL allowESIMAutoFillForWebKit(id, SEL, NSString *host, NSError **)
+    [navigationDelegate allowAnyTLSCertificate];
+
+    RetainPtr configuration = server.httpsProxyConfiguration();
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500) configuration:configuration.get()]);
```
I think you missed this one!

aprotyas reviewed (commented):
Do you mind flipping the order of the bugzilla and radar links? Our usual commit log format prefers bugzilla first.

Also, if you could squash all the commits and force push to the branch again, I can put you on the merge queue.

aprotyas reviewed (commented):
Another editorial comment, if you could retitle the commit to say something like "Add test coverage for ..."?

arenevier committed:
Add test coverage for: Add certificate info to WKFrameInfo accessor of _WKFocusedElementInfo.
https://bugs.webkit.org/show_bug.cgi?id=297061
rdar://157669975

Reviewed by Abrar Rahman Protyasha.

Api test checking that in
setFocusRequiresStrongPasswordAssistanceHandler callback, _serverTrust
is set correctly in _WKFocusedElementInfo.frame.

* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::focusedElementInformation):
* Tools/TestWebKitAPI/Tests/ios/KeyboardInputTestsIOS.mm:
(TestWebKitAPI::TEST(KeyboardInputTests, TestFrameInfoServerTrustDuringStrongPasswordAssistance)):

Canonical link: https://commits.webkit.org/298455@main
